Extensions/blacklist.js

```diff
@@ -707,36 +746,53 @@ XKit.extensions.blacklist = new Object({
 					'<div data-post-id="' + $(obj).attr('id') + '" class="xblacklist_open_post xkit-button">Show it anyway</div></div>';
 		}
 
-		$(obj).addClass("xblacklist_blacklisted_post");
 		$(obj).find(".post_info").css("display","none");
 		$(obj).find(".post_controls").css("display","none");
 		$(obj).find(".post_content").html(old_content + block_excuse);
 		$(obj).find(".post_footer_links").css('display','none');
 		$(obj).find(".post_source").css('display','none');
 		$(obj).find(".post-source-footer").css('display','none');
+        $(obj).find(".post_answer").css("display","none");
+    },
 
-		if (XKit.extensions.blacklist.preferences.mini_block.value !== true) {
-			$(obj).addClass("xblacklist_blacklisted_post_full_ui");
-		}
+    hide_mobile_content: function(obj, word) {
+        var to_add_type = "";
 
-		if (XKit.extensions.blacklist.preferences.show_tags.value === true && XKit.extensions.blacklist.preferences.mini_block.value === false) {
-			$(obj).find(".post_footer").css('display','none');
-		} else {
-			$(obj).find(".post_tags, .post_footer").css('display','none');
-		}
+		if (XKit.extensions.blacklist.preferences.show_type.value === true) {
+            var post_type = "";
+            if ($(obj).hasClass("post_type_photo")) {
+                post_type = "photo";
+            } else if ($(obj).hasClass("post_type_photoset")) {
+                post_type = "photoset";
+            } else if ($(obj).hasClass("post_type_note")) {
+                post_type = "note";
+            } else if ($(obj).hasClass("post_type_regular")) {
+                post_type = "regular";
+            } else if ($(obj).hasClass("post_type_video")) {
+                post_type = "video";
+            }
+			to_add_type = "<div class=\"xkit-blacklist-post-type  " + post_type + "\">&nbsp;</div>";
 
-		if ($(obj).hasClass("xkit-shorten-posts-shortened") === true) {
+		}
 
-			// This was also shortened.
-			$(obj).attr('data-xkit-blacklist-old-height', $(obj).css("height"));
-			$(obj).css("height","auto");
-			$(obj).find(".xkit-shorten-posts-embiggen").css("display","none");
+        var block_excuse = '<div class="xblacklist_excuse">' +
```

Comment:
these two block_excuse templates are shared between mobile and desktop, should be refactored out.